test/metahelper/util/util.go

```diff
@@ -0,0 +1,97 @@
+// Package util supports various needs for running tests
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+
+	// TODO(chaodaiG): use prow from pkg once
+	// https://github.com/knative/pkg/pull/750 is merged
+	"knative.dev/test-infra/shared/prow"
+)
+
+const (
+	Filename = "metadata.json"
+)
+
+// client holds metadata as a string:string map, as well as path for storing
+// metadata
+type client struct {
+	MetaData map[string]string
+	Path     string
+}
+
+// NewClient creates a client, takes custom directory for storing `metadata.json`.
+// It reads existing `metadata.json` file if it exists, otherwise creates it.
+// Errors out if there is any file i/o problem other than file not exist error.
+func NewClient(dir string) (*client, error) {
+	c := &client{
+		MetaData: make(map[string]string),
+	}
+	if dir == "" {
+		log.Println("Getting artifacts dir from prow")
+		dir = prow.GetLocalArtifactsDir()
+	}
+	c.Path = path.Join(dir, Filename)
+	_, err := os.Stat(dir)
+	if err != nil && os.IsNotExist(err) {
+		err = os.MkdirAll(dir, 0777)
+	}
+
+	return c, err
+}
+
+// sync reads from meta file and convert it to Meta, returns empty
+// Meta if file doesn't exist yet, and returns error if there is any i/o or
+// unmarshall error
+func (c *client) sync() error {
+	_, err := os.Stat(c.Path)
+	if os.IsNotExist(err) {
+		log.Println("write file")
```

Comment:
If this is debug, remove. If not, this message should be more useful, like showing the location of the file.